packages/lit-virtualizer/src/lib/repeat.ts

```diff
@@ -0,0 +1,147 @@
+import { directive, NodePart, createMarker, TemplateResult } from 'lit-html';
+import { VirtualRepeater } from './uni-virtualizer/lib/VirtualRepeater';
+import { Layout } from './uni-virtualizer/lib/layouts/Layout';
+
+/**
+ * Mixin for VirtualRepeater and VirtualScroller. This mixin overrides the generic
+ * methods in those classes to provide lit-specific implementations of element
+ * creation and manipulation.
+ * 
+ * This mixin implements child recycling, so children can be reused after removal
+ * from the DOM.
+ */
+export const LitMixin = Superclass => class extends Superclass {
+  // NodeParts that are available for reuse.
+  _pool: Array<NodePart>;
+
+  // Method for generating each item's DOM.
+  _template: (item: any, index?: number) => TemplateResult;
+
+  // Children are rendered into this part.
+  // The host of the directive that constructs the scroller.
+  _hostPart: NodePart;
+  
+  /**
+   * TODO @straversi: This is not a true mixin. Mixins do have information
+   * about potential superclasses. This class appears to know the constructor
+   * arguments of an unknown superclass. It also overrides methods on an
+   * unknown superclass.
+   */
+  constructor(config: {
+    part: NodePart,
+    template: (item: any, index?: number) => TemplateResult,
+    useShadowDOM?: boolean,
+    scrollTarget?: Element | Window,
+    layout?: Layout
+  }) {
+    const {part, template, useShadowDOM, layout} = config;
+    let container = part.startNode.parentNode;
+    let scrollTarget = config.scrollTarget || container;
+    super({container, scrollTarget, useShadowDOM, layout});
+
+    this._pool = [];
+    this._template = template;
+    this._hostPart = part;
+  }
+
+  createElement(): NodePart {
```

Comment:
Why does `createElement` return a `NodePart`? Seems like odd naming. Can we standardized on "item" for... items?

Reply:
Currently, we call the members of the data array "items" and the DOM representation of those items "elements".

Here, we're overriding the vanilla implementation (which deals with elements directly) with a lit-specific implementation (which deals with NodeParts, which in turn yield elements).